bib/cmd/bootc-image-builder/image.go

```diff
@@ -293,27 +195,97 @@ func manifestForISO(c *ManifestConfig, rng *rand.Rand) (*manifest.Manifest, erro
 				ImageFormat: platform.FORMAT_ISO,
 			},
 			BIOS:       true,
-			UEFIVendor: "fedora",
+			UEFIVendor: c.SourceInfo.UEFIVendor,
 		}
 	case arch.ARCH_AARCH64:
+		// aarch64 always uses UEFI, so let's enforce the vendor
+		if c.SourceInfo.UEFIVendor == "" {
+			return nil, fmt.Errorf("UEFI vendor must be set for aarch64 ISO")
+		}
 		img.Platform = &platform.Aarch64{
 			BasePlatform: platform.BasePlatform{
 				ImageFormat: platform.FORMAT_ISO,
 			},
-			UEFIVendor: "fedora",
+			UEFIVendor: c.SourceInfo.UEFIVendor,
 		}
 	}
 
 	img.OSName = "default"
 	img.Filename = "install.iso"
 
 	mf := manifest.New()
-	mf.Distro = manifest.DISTRO_FEDORA
-	runner := &runner.Fedora{Version: 39}
-	_, err := img.InstantiateManifest(&mf, c.Repos, runner, rng)
+
+	foundDistro, foundRunner, err := getDistroAndRunner(c.SourceInfo.OSRelease)
+	if err != nil {
+		return nil, fmt.Errorf("failed to infer distro and runner: %w", err)
+	}
+	mf.Distro = foundDistro
+
+	_, err = img.InstantiateManifest(&mf, nil, foundRunner, rng)
 	return &mf, err
 }
 
+func getDistroAndRunner(osRelease source.OSRelease) (manifest.Distro, runner.Runner, error) {
```

Comment:
(super nitpick) maybe `distrodef` is a good place for this helper? It feels a bit misplaced in here :) (but fine for later)

Reply:
I think this should actually belong to the `images` library. WDYT?